File: src/frontend/src/types/notification.ts
export type NotificationSeverity = 'info' | 'warning' | 'critical';

// Legacy type used by gamification store
export interface NotificationItem {
  id: string;
  title: string;
  body: string;
  severity: NotificationSeverity;
  read: boolean;
  created_at: string;
}

// Sprint 5: API notification type
export type NotificationType = 'achievement' | 'alert' | 'system' | 'collaboration';

export interface Notification {
  id: string;
  type: NotificationType;
  title: string;
  body: string | null;
  read: boolean;
  link: string | null;
  created_at: string;
}

File: src/backend/app/models/notification.py
from datetime import datetime
from typing import Optional
from sqlalchemy import Boolean, ForeignKey, String, Text, Index
from sqlalchemy.orm import Mapped, mapped_column
import uuid

from app.core.database import Base
from app.core.utils import utcnow


class Notification(Base):
    __tablename__ = "notifications"
    __table_args__ = (
        Index("ix_notifications_user_read", "user_id", "read"),
        Index("ix_notifications_workspace_id", "workspace_id"),
    )

    id: Mapped[str] = mapped_column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    workspace_id: Mapped[str] = mapped_column(ForeignKey("workspaces.id", ondelete="CASCADE"), nullable=False)
    user_id: Mapped[str] = mapped_column(ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    type: Mapped[str] = mapped_column(String(50), nullable=False)  # achievement, alert, system, collaboration
    title: Mapped[str] = mapped_column(String(255), nullable=False)
    body: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    read: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    link: Mapped[Optional[str]] = mapped_column(String(500), nullable=True)
    created_at: Mapped[datetime] = mapped_column(default=utcnow)
